docs/setup_devcontainer_environment.mdx

```diff
@@ -0,0 +1,130 @@
+---
+title: "Setting Up a Devcontainer Development Environment"
+---
+
+The easiest way to get started with development is to use Visual Studio Code with [devcontainers](https://code.visualstudio.com/docs/devcontainers/containers). This approach creates a preconfigured development environment with the necessary tools. This approach is enabled for all Home Assistant repositories. Since the Docker-based devcontainer concept provides a layer of abstraction, the following guides merge at one point.
+
+First, install [Visual Studio code](https://code.visualstudio.com/), next follow one of the next chapters based on your system.
+
+## Setup on Windows
+
+:::tip
+**Quick start guide**
+
+- Install [WSL2](https://learn.microsoft.com/en-us/windows/wsl/install) and enable [systemd support](https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/).
+- Install the [Docker engine](https://docs.docker.com/engine/install/) based on your selected WSL2 Linux distribution.
+- Fork and clone the desired code repository.
```

Comment:
Should it be documented to clone the desired code within WSL?
I think there are performance differences if the repo is cloned in Windows and not inside WSL.

Reply:
yeah you're right ... for me it is common to do it within the wsl2, but for new developers it might not be that obvious.